crates/matrix-sdk/src/account.rs

```diff
@@ -770,18 +819,33 @@ impl Account {
         // Now we need to mark the room as a DM for ourselves, we fetch the
         // existing `m.direct` event and append the room to the list of DMs we
         // have with this user.
-        let mut content = self
-            .account_data::<DirectEventContent>()
-            .await?
-            .map(|c| c.deserialize())
-            .transpose()?
+
+        // TODO: Uncomment this once we can rely on `/sync`. Read further bellow for a
+        // complete explanation.
+        // let mut content = self
+        //     .account_data::<DirectEventContent>()
+        //     .await?
+        //     .map(|c| c.deserialize())
+        //     .transpose()?
+        //     .unwrap_or_default();
+
+        // We are fetching the content from the server because we currently can't rely
+        // on `/sync` giving us the correct data in a timely manner.
+        //
+        // Because of the `unwrap_or_default()` we're using, this may lead to us
+        // resetting the `m.direct` account data event and unmarking the user's
+        // DMs.
```

Comment:
But now that we use `fetch_account_data` we should be guaranteed to have the latest event, if the server responds with a successful response at all?

Reply:
Yeah, this was supposed to explain why we are using `fetch_account_data()` instead of the above. I guess I should make this clearer by moving things to the right places.

Reply:
Hopefully made this clearer now.